src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -246,6 +249,52 @@ void QueryToOperatorTransformer::Visit(parser::InsertStatement *op) {
     insert_expr->PushChild(output_expr_);
     output_expr_ = insert_expr;
   } else {
+    auto storage_manager = storage::StorageManager::GetInstance();
+    auto database =
+        storage_manager->GetDatabaseWithOid(target_table->GetDatabaseOid());
+    auto table = database->GetTableWithOid(target_table->GetTableOid());
+
+    auto *schema = table->GetSchema();
```

Comment:
You shouldn't be accessing the physical table, but rather access the catalog to retrieve table schema information. Look at how `parser::TableRef` is handled.

Reply:
You mean `TableRef` in the `Visit()` method?

Reply:
I went through `parser::TableRef`'s source but it doesn't have any getter method for table schema.